src/data/tournament.py

```diff
@@ -440,14 +441,8 @@ def players_by_name_without_unpaired(self) -> list[Player]:
 
     @cached_property
     def players_by_rank(self) -> dict[int, Player]:
-        ranked_players = sorted(
-            self.players_by_id.values(),
-            key=lambda player: player.rank_sort_key,
-        )
-        return {
-            rank: player for rank, player in
-            enumerate(ranked_players, start=1)
-        }
+        assert self._players_by_rank is not None, 'Tournament._players_sorted_by_rank is not set, call Tournament.compute_player_ranks() before.'
+        return self._players_by_rank
```

Comment:
Maybe compute the ranks, instead of throwing an `AssertionError`?
If you're not computing anything, this method can be a property.